### DISCUSSION TITLE
Allow to have shadow dom content inside editor

### DISCUSSION THREAD

**Discussion by ufoloko** [GrapesJS/grapesjs#6275] [Ideas]
Sometimes you use grapejs editor, to modify some partial html content on a page or CMS, and sometimes you use some generic stiles for example container, row, etc, from bootstrap, but if you change some property like color, or font-size and then apply to selected styles, you will modify style for all page using those styles.

It Should be great to have some option to be able to save styles from editor in Dom Shadow node, so, if you add/modify some style this change don´t reflect on the rest of contents.

**Comment by artf** [comment]
Check this https://grapesjs.com/docs/modules/Selectors.html#component-first-selectors
